sig-scalability/configs-and-limits/thresholds.md

```diff
@@ -2,100 +2,73 @@
-| Number of deployments per ns        | 2000           |             | 10000          |
-| Number of pods per deployment       | TBD            |             | 10000          |
-| Number of jobs per ns               | TBD            |             | 1000           |
-| Number of daemon sets per ns        | TBD            |             | 100            |
-| Number of stateful sets per ns      | TBD            |             | 100            |
-| Number of secrets per ns            | TBD            |             | TBD            |
-| Number of secrets per pod           | TBD            |             | TBD            |
-| Number of config maps per ns        | TBD            |             | TBD            |
-| Number of config maps per pod       | TBD            |             | TBD            |
-| Number of storageclasses            | TBD            |             | TBD            |
-| Number of roles and rolebindings    | TBD            |             | TBD            |
+There are couple caveats to the thresholds we are presenting below:
+1. In majority of cases, thresholds are NOT hard limits - crossing
+   the limit results in degraded performance and doesn't mean cluster
+   immediately fails over.
+1. **Many of the tresholds (for cluster scope) are given for the largest
+   possible cluster. For smaller clusters, the limits are proportionally
+   lower.**
+1. The thresholds may differ (hopefully be non-decreasing) across Kubernetes
+   releases. The threshold below are given for Kubernetes head. <br/>
+   **TODO:** We are planning to start versioning the table below, but we
+   are not there yet.
+1. Given that configuration influences thresholds, we are assuming vanilla
+   Kubernetes setup.
 
-There are also thresholds for other types, but for those the numbers depend
-also on the environment (bare metal or which cloud provider) the cluster is
-running in. These include:
+The table below is **NOT exhaustive** - more content is coming soon.
 
-| Quantity                                  | Head threshold | 1.8 release | Long term goal |
-|-------------------------------------------|----------------|-------------|----------------|
-| Number of ingresses                       | TBD            |             | TBD            |
-| Number of PersistentVolumes               | TBD            |             | TBD            |
-| Number of PersistentVolumeClaims per ns   | TBD            |             | TBD            |
-| Number of PersistentVolumeClaims per node | TBD            |             | TBD            |
+| Quantity               | Threshold scope=namespace | Threshold: scope=cluster |
+|------------------------|---------------------------|--------------------------|
+| #Nodes                 | n/a                       | 5000                     |
+| #Namespaces            | n/a                       | 10000                    |
+| #Pods                  | 3000                      | 150000                   |
```

Comment:
@wojtek-t Hi, I'm a bit confused about the namespaced pods threshold 3000. We already have a threshold on node 5000, so there would be some cases that exceed the namespaced pods threshold, for example, kube-proxy deployed as daemonset, there would be 5k pods in kube-system in a cluster with 5k nodes.

So, why we set the threshold at 3000?

Reply:
This is a good point. This was set to max value we were (and still are) testing, but ignoring the kube-system namespace. We could probably bump this up.

Reply:
Got it, Thanks a lot. Actually, in my production, there have been 5k nodes cluster and and deployed several daemonsets in one namespace in which there are more than 10k pods. So, I'm considering if such case is a potential risk against stability of my production cluster.

Reply:
I see. I think what you risk is more some slowness of controllers (reconciliation in many controllers process is proportional to size of the namespace wrt objects they are interested in). But i wouldn't say there is a significant stability risk there.

Reply:
Thanks a lot, now I am more confident of my production with 5k nodes.